### ISSUE TITLE
FTL Offline (Dev)

### ISSUE THREAD

**Issue by kyle95wm** [pi-hole/pi-hole#2144]
**In raising this issue, I confirm the following:** `{please fill the checkboxes, e.g: [X]}`

- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [X] The issue I am reporting can be *replicated*.
- [X] The issue I am reporting isn't a duplicate (see [FAQs](https://github.com/pi-hole/pi-hole/wiki/FAQs), [closed issues](https://github.com/pi-hole/pi-hole/issues?utf8=%E2%9C%93&q=is%3Aissue%20is%3Aclosed%20), and [open issues](https://github.com/pi-hole/pi-hole/issues)).

**How familiar are you with the the source code relevant to this issue?:**

1
---
**Expected behaviour:**

FTL being online after switching to dev branch to beta test FTLDNS

**Actual behaviour:**

FTL is "Offline"

**Steps to reproduce:**

Clean install then switch to dev branch by running "pihole checkout dev"

**Debug token provided by [uploading `pihole -d` log](https://discourse.pi-hole.net/t/the-pihole-command-with-examples/738#debug):**

19fmx1dfvc

**Troubleshooting undertaken, and/or other relevant information:**

- Restarted the server
- Restarted pihole's components
- Hard refreshed browser cache
- Switching from dev to master, and back again

**Comment by technicalpyro** [comment]
is the file `/etc/pihole/ftlbranch` present? and if so what are its contents?

**Comment by kyle95wm** [comment]
Yes, and the contents:

development

**Comment by Mcat12** [comment]
Run `pihole -v` and share the output.

**Comment by kyle95wm** [comment]
Pi-hole version is v3.3-152-g05d8a7f (Latest: v3.3.1)
  AdminLTE version is v3.2.1-169-g70c5fa1 (Latest: v3.3)
  FTL version is vDev-1b4be77 (Latest: v3.0)

Here's another token: nqfooxjkwk

To reproduce this, I ran
`pihole checkout dev`

**Comment by fxhx** [comment]
Same here
  Pi-hole version is v3.3-152-g05d8a7f (Latest: v3.3.1)
  AdminLTE version is v3.2.1-169-g70c5fa16 (Latest: v3.3)
  FTL version is vDev-1b4be77 (Latest: v3.0)

Token: abdmht8es9

**Comment by p1r473** [comment]
This fixed it for me

sudo rm /usr/bin/pihole-FTL
pihole -r

**Comment by Mcat12** [comment]
@kyle95wm It looks like dnsmasq is still running. Disable it via `systemctl`

**Comment by Mcat12** [comment]
@fxhx It looks like you are having the same issue as @kyle95wm

**Comment by kyle95wm** [comment]
This should be a fix in the dev branch so that one doesn't have to disable dnsmasq, if this is the fix. I'll have to try it out next chance I get.

**Comment by kyle95wm** [comment]
@p1r473 your fix worked for me. dnsmasq is disabled automatically and FTL is running. I can confirm this via the web interface

Here is the command I ran:
`sudo rm -rf /usr/bin/pihole-FTL && pihole -r`

and then clicked "Repair" and all is good. Here's a token so that the devs have more info:

t8z71ooch9

```user@pihole-vm:~$ pihole -v
  Pi-hole version is v3.3-152-g05d8a7f (Latest: v3.3.1)
  AdminLTE version is v3.2.1-169-g70c5fa1 (Latest: v3.3)
  FTL version is vDev-1b4be77 (Latest: v3.0)````

**Comment by kyle95wm** [comment]
Just an update:

Even to this day, check out to the development branch REFUSES to auto-disable dnsmasq. What gives?

**Comment by Mcat12** [comment]
Should be fixed by #2159

**Comment by kyle95wm** [comment]
Yep. Can confirm. The PR references #2156
